### PR TITLE
modified configuremake and cmakemake to allow changing configure_cmd,…

### PR DESCRIPTION
**Pull request by mboisson**
… build_cmd and install_cmd to fit various needs

This is for what we discussed over Slack, to allow to change, for example, "make install" by "ninja install" in one case. 

Tagging @boegel  and @akesandgren

### CODE REVIEW

File: easybuild/easyblocks/generic/cmakemake.py [hunk 1]
```diff
@@ -107,7 +109,8 @@ def configure_step(self, srcdir=None, builddir=None):
 
         options_string = ' '.join(options)
 
-        command = "%s cmake %s %s %s" % (self.cfg['preconfigopts'], srcdir, options_string, self.cfg['configopts'])
+        command = "%s %s %s %s %s" % (self.cfg['preconfigopts'], self.cfg['configure_cmd'], srcdir, options_string,
+                                      self.cfg['configopts'])
```
Comment (boegel) [inline easybuild/easyblocks/generic/cmakemake.py]: @mboisson Do you mind changing this to:

```python
command = ' '.join([
    self.cfg['preconfigopts'],
    self.cfg['configure_cmd'],
    srcdir,
    options_string,
    self.cfg['configopts'],
])
```

(similar below, but you can make it a single line if it's short enough)

**Comment by boegel** [comment]
@mboisson Looks like you have a merge conflict to fix as well...

**Comment by mboisson** [comment]
> @mboisson Looks like you have a merge conflict to fix as well...

I merged easybuilders-develop into this branch.

File: easybuild/easyblocks/generic/cmakemake.py [hunk 2]
```diff
@@ -109,7 +111,12 @@ def configure_step(self, srcdir=None, builddir=None):
 
         options_string = ' '.join(options)
 
-        command = ' '.join([self.cfg['preconfigopts'], 'cmake', options_string, self.cfg['configopts'], srcdir])
+        command = ' '.join([
+            self.cfg['preconfigopts'],
+            self.cfg['configure_cmd'],
```
Comment (boegel) [inline easybuild/easyblocks/generic/cmakemake.py]: @mboisson We should be a bit more careful here, and use `self.cfg.get`, see https://github.com/ComputeCanada/easybuild-easyblocks/pull/2

**Comment by mboisson** [comment]
merged, but that created conflicts

**Comment by boegel** [comment]
@mboisson Actually, the conflict was already there before you merged my PR...

**Comment by mboisson** [comment]
Mmmm, it wasn't there 5 days ago, I merged easybuilders/develop into this branch 5 days ago

**Comment by boegel** [comment]
@mboisson conflict was caused by #1652, I've fixed it now for this PR...

**Comment by boegel** [comment]
Going in, thanks @mboisson!

**Pull request merged**